backend/service/chaos/experimentation/experimentstore/experiment_store_test.go

```diff
@@ -169,74 +169,72 @@ func TestCancelExperimentRun(t *testing.T) {
 	}
 }
 
-var getExperimentsTests = []struct {
-	id   string
-	sql  string
-	args []driver.Value
-	rows [][]driver.Value
-	err  error
-}{
-	{
-		id:   "get all experiments",
-		sql:  `SELECT experiment_run.id, details FROM experiment_config, experiment_run WHERE experiment_config.id = experiment_run.experiment_config_id AND ($1 = '' OR $1 = experiment_config.details ->> '@type')`,
-		args: []driver.Value{"upstreamCluster", "downstreamCluster", 1},
-		rows: [][]driver.Value{
-			{
-				1234,
-				`{"@type": "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig","clusterPair":{"downstreamCluster":"upstreamCluster","upstreamCluster":"downstreamCluster"},"abort":{"percent":100,"httpStatus":401}}`,
-			},
-		},
-	},
-}
+var getExperimentsSQLQuery = `SELECT experiment_run.id, details FROM experiment_config, experiment_run WHERE experiment_config.id = experiment_run.experiment_config_id AND ($1 = '' OR $1 = experiment_config.details ->> '@type')`
 
-func TestGetExperiments(t *testing.T) {
-	t.Parallel()
+func TestGetExperimentsUnmarshalsExperimentConfiguration(t *testing.T) {
+	assert := assert.New(t)
 
-	for _, test := range getExperimentsTests {
-		test := test
+	db, mock, err := sqlmock.New()
+	assert.NoError(err)
 
-		t.Run(test.id, func(t *testing.T) {
-			t.Parallel()
-			a := assert.New(t)
+	es := &experimentStore{db: db}
+	defer es.Close()
 
-			db, mock, err := sqlmock.New()
-			a.NoError(err)
+	expected := mock.ExpectQuery(regexp.QuoteMeta(getExperimentsSQLQuery)).WithArgs("foo")
+	rows := sqlmock.NewRows(experimentColumns)
+	rows.AddRow([]driver.Value{
+		1234,
+		`{"@type": "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig","clusterPair":{"downstreamCluster":"upstreamCluster","upstreamCluster":"downstreamCluster"},"abort":{"percent":100,"httpStatus":401}}`,
+	}...)
+	expected.WillReturnRows(rows)
 
-			es := &experimentStore{db: db}
-			defer es.Close()
+	experiments, err := es.GetExperiments(context.Background(), "foo")
+	assert.NoError(err)
 
-			expected := mock.ExpectQuery(regexp.QuoteMeta(test.sql)).WithArgs("foo")
-			if test.err != nil {
-				expected.WillReturnError(test.err)
-			} else {
-				rows := sqlmock.NewRows(experimentColumns)
-				for _, row := range test.rows {
-					rows.AddRow(row...)
-				}
-				expected.WillReturnRows(rows)
-			}
+	assert.Equal(1, len(experiments))
+	experiment := experiments[0]
+	assert.Equal(uint64(1234), experiment.GetId())
 
-			experiments, err := es.GetExperiments(context.Background(), "foo")
-			if test.err != nil {
-				a.Equal(test.err, err)
-			} else {
-				a.NoError(err)
-			}
+	config := &serverexperimentation.TestConfig{}
+	err2 := ptypes.UnmarshalAny(experiment.GetConfig(), config)
+	if err2 != nil {
+		t.Errorf("setSnapshot failed %v", err2)
+	}
```

Comment:
Is this different from `assert.NoError(err2)`?

Reply:
it should be similar - let me check

Reply:
updated, thanks for the comment!